github/store/db.go

```diff
@@ -47,72 +47,180 @@ const (
 )
 
 var tables = []string{
-	"organizations_versioned",
-	"users_versioned",
-	"repositories_versioned",
-	"issues_versioned",
-	"issue_comments_versioned",
-	"pull_requests_versioned",
-	"pull_request_reviews_versioned",
-	"pull_request_comments_versioned",
+	"github_organizations_versioned",
+	"github_users_versioned",
+	"github_repositories_versioned",
+	"github_issues_versioned",
+	"github_issue_comments_versioned",
+	"github_pull_requests_versioned",
+	"github_pull_request_reviews_versioned",
+	"github_pull_request_comments_versioned",
 }
 
 func (s *DB) SetActiveVersion(ctx context.Context, v int) error {
-	// TODO: for some reason the normal parameter interpolation $1 fails with
-	// pq: got 1 parameters but the statement requires 0
+	// Unified schema
 
-	_, err := s.DB.ExecContext(ctx, fmt.Sprintf(`CREATE OR REPLACE VIEW organizations AS
+	_, err := s.DB.ExecContext(ctx, "DROP MATERIALIZED VIEW IF EXISTS owners")
+	if err != nil {
+		return fmt.Errorf("failed to drop unified view owners: %v", err)
+	}
+	_, err = s.DB.ExecContext(ctx, fmt.Sprintf(`CREATE MATERIALIZED VIEW owners AS
+	SELECT login, name
+	FROM github_organizations_versioned WHERE %v = ANY(versions)`, v))
+	if err != nil {
+		return fmt.Errorf("failed to create unified view owners: %v", err)
+	}
```

Comment:
I was thinking about something like the example below.
Imo, that way it could be reviewed more easily, and it could be also useful for documentation purposes because it will be more clear the mapping between the view, and its underlying data.

```go
var views map[string]string = map[string]string{
  "owners": `SELECT login, name
￼	FROM github_organizations_versioned WHERE %v = ANY(versions)`,
  "users": `SELECT login, name
￼	FROM github_users_versioned WHERE %v = ANY(versions)`,
  "issues": `SELECT repository_owner, repository_name, repository_owner || '/' || repository_name AS repository_full_name,
￼           number, state, title, body,
￼           created_at, closed_at, updated_at, comments, user_id, user_login, htmlurl AS html_url, labels
￼	FROM github_issues_versioned WHERE %v = ANY(versions)`,
}

func createMaterializedVire(viewname string, viewFilter, version int) err {
  _, err := s.DB.ExecContext(ctx, fmt.Printf("DROP MATERIALIZED VIEW IF EXISTS %s", viewname))
  if err != nil {
    return fmt.Errorf("failed to drop unified view %s: %v", viewname, err)
  }

  _, err = s.DB.ExecContext(ctx, fmt.Sprintf(`CREATE MATERIALIZED VIEW %s AS %s`,
    fmt.Printf(viewFilter, version),
    viewname,
  ))
  if err != nil {
    return fmt.Errorf("failed to create unified view %s: %v", viewname, err)
  }

  return nil
}

func (s *DB) SetActiveVersion(ctx context.Context, v int) error {
  for name, query := range views {
    if err := createMaterializedVire(name, query, v); err != nil {
      return err
    }
  }

  return nil
}
```

Reply:
yup, if we can do something like this I think it would more readable, instead of having a wall of sql statements.

Reply:
if I change to a map it would still be a wall of sql queries. Let me think about how to improve it.

Reply:
I couldn't come but with anything better than this: https://github.com/src-d/metadata-retrieval/pull/45/commits/59c1805f1026f51b9ee541e1b5bf1dd92d7f5605

Reply:
mhmmm... doesn't change that much in terms of _visual perception_ 😂. I'm ok in both ways.

Reply:
I really prefer this other version, because now the SQL code and logic are clearly separated.
thanks!